### PR TITLE
[FEATURE] General reusable validation implementation and middleware for validating DOMDocument

### PR DESCRIPTION
**Pull request by markusweigelt**
This PR provides abstract classes to derived from for custom validation and reusable implementations for validation of DOMDocument.
- Abstract validator and validation stack from which it can be derived (Can be handled as standalone validators as well, since they are derived from the `AbstractValidator` of Extbase)
- Implementation and tests of `DOMDocumentValidationStack`, `XMLSchemesValidator` and `SaxonXslToSvrlValidator`
- Comprehensive documentation
- Customisable `DOMDocumentValidation` middleware via TypoScript setting `domDocumentValidationValidators`

**Example of middleware**

URI Query: `dlf/domDocumentValidation&url=https%3A%2F%2Fwww.example.com%2Fmetsmods.xml`

JSON response
```
{
  "valid": false,
  "results": {
    "XML-Schemes Validator": [
      "Element '{http://www.loc.gov/METS/}mdWrap', attribute 'MDTYP': The attribute 'MDTYP' is not allowed.\n",
      "Element '{http://www.loc.gov/METS/}mdWrap': The attribute 'MDTYPE' is required but missing.\n"
    ],
    ...
  }
}
```

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -84,14 +84,12 @@ Build/Test/runTests.sh -w  # Watch mode
...
 ```bash
-Build/Test/runTests.sh -s functional
 Build/Test/runTests.sh -s functional -w  # Watch mode
 ```
 
 To learn about available options (e.g., to select the PHP version), check the usage info:
 
 ```bash
-Build/Test/runTests.sh -h
 ```
```
Comment (sebastian-meyer) [inline DEVELOPMENT.md]: Without the removed command this makes no more sense. Was the removal intentional? Then this should be removed as well.

**Review by sebastian-meyer — CHANGES_REQUESTED** [review]

**Comment by markusweigelt** [comment]
@sebastian-meyer Thanks, I had overlooked that. I added the lines back.

**Review by sebastian-meyer — APPROVED** [review]

**Comment by thomaslow** [comment]
Hi Sebastian and Markus. Your recent changes fail unit tests in my branch (php 7.4, typo3 v11). Any ideas?

```
There was 1 error:

1) Kitodo\Dlf\Tests\Unit\Validation\SaxonXslToSvrlValidatorTest::testValidation
InvalidArgumentException: Saxon JAR file not found.

/home/runner/work/kitodo-presentation/kitodo-presentation/Classes/Validation/SaxonXslToSvrlValidator.php:47
/home/runner/work/kitodo-presentation/kitodo-presentation/Tests/Unit/Validation/SaxonXslToSvrlValidatorTest.php:67
phpvfscomposer:///home/runner/work/kitodo-presentation/kitodo-presentation/vendor/phpunit/phpunit/phpunit:106
/home/runner/work/kitodo-presentation/kitodo-presentation/vendor/bin/phpunit:118

--

There was 1 failure:

1) Kitodo\Dlf\Tests\Unit\Validation\SaxonXslToSvrlValidatorTest::testXslFileNotFound
Failed asserting that exception message 'Saxon JAR file not found.' contains 'XSL Schematron file not found.'.

phpvfscomposer:///home/runner/work/kitodo-presentation/kitodo-presentation/vendor/phpunit/phpunit/phpunit:106
/home/runner/work/kitodo-presentation/kitodo-presentation/vendor/bin/phpunit:118
```

**Comment by markusweigelt** [comment]
@thomaslow The problem seem to be the Typo3 function `GeneralUtility::getFileAbsFileName` https://github.com/thomaslow/kitodo-presentation/blob/6363217ab3c7c0a1863e45dfe75de2ebe16b7949/Classes/Validation/SaxonXslToSvrlValidator.php#L43, cause with extension path `EXT:dlf/Tests/Fixtures/Format/alto.xml` parameter the resolved path is not returned when running with the updated `typo3/testing-framework`. The behavior of resolving extension paths may have changed in the new version.

**Comment by markusweigelt** [comment]
@thomaslow i run the tests again with `typo3/testing-framework` `6.16.9`, which succeeds, and with `7.1.0` I can reproduce the problem. I'll take a look at it tomorrow when I get a chance.

**Comment by thomaslow** [comment]
@markusweigelt I followed the empty string of `GeneralUtility::getFileAbsFileName` to the Typo3 [PackageManager->getActivePackages()](https://github.com/TYPO3/typo3/blob/main/typo3/sysext/core/Classes/Package/PackageManager.php#L621). It seems the package key `EXT:dlf` can not be resolved, because the package is not known at all when running the unit test. 

The difference between `testing-framework` v6 and v7 seems to be in the [UnitTestsBootstrap.php](https://github.com/TYPO3/testing-framework/compare/6...7#diff-1557a22f50b51bb776e57ff5f503802e3e5a668dab6f590c9b5236e016a9291a). They set `composerMode = true`, which prevents the [PackageManager](https://github.com/TYPO3/typo3/blob/2126407004b2dec61a14dd0059655f43f185a00a/typo3/sysext/core/Classes/Package/PackageManager.php#L247C1-L247C2) from scanning for extensions. Removing this check for composer mode loads all package paths including `dlf` and the unit test works.

I'm not sure how to fix this. I see two options:
- not use `GeneralUtility::getFileAbsFileName` with an extension key in unit tests, e.g., specify only absolute file paths
- move tests to functional tests, which have a different bootstrap process

**Comment by sebastian-meyer** [comment]
We copied `Build/Test/runTests.sh` a while ago from TYPO3, maybe we need to update this script in order to correctly run the tests in `composerMode`? The script seems really old anyways, referencing a lot of outdated versions (i. e. for MySQL) and missing current versions (like PHP 8.3).

**Comment by sebastian-meyer** [comment]
I opened a new issue in order to discuss this further: #1348

**Pull request merged**